src/components/timecard/edit-shift-hours/EditShiftHours.spec.jsx

```diff
@@ -530,6 +534,104 @@ describe('EditShiftHours', () => {
     expect(message).toBeTruthy();
   });
 
+  it('should show start and end dates when checking box', async () => {
+    renderWithTimecardContext(
+      <EditShiftHours
+        setShowEditShiftHours={jest.fn()}
+        timeEntry={newTimeEntry}
+        timeEntriesIndex={0}
+      />,
+      defaultTimecardContext
+    );
+
+    act(() => {
+      const checkBox = screen.getByText('View or edit dates');
+      fireEvent.click(checkBox);
+    });
+
+    const startDay = screen.getByTestId(testInputNames.startDay);
+    const startDayValue = startDay.getAttribute('value');
+    expect(startDayValue).toEqual('01');
+    const startMonth = screen.getByTestId(testInputNames.startMonth);
+    const startMonthValue = startMonth.getAttribute('value');
+    expect(startMonthValue).toEqual('09');
+    const startYear = screen.getByTestId(testInputNames.startYear);
+    const startYearValue = startYear.getAttribute('value');
+    expect(startYearValue).toEqual('2022');
+
+    const endDay = screen.getByTestId(testInputNames.endDay);
+    const endDayValue = endDay.getAttribute('value');
+    expect(endDayValue).toEqual('01');
+    const endMonth = screen.getByTestId(testInputNames.endMonth);
+    const endMonthValue = endMonth.getAttribute('value');
+    expect(endMonthValue).toEqual('09');
+    const endYear = screen.getByTestId(testInputNames.endYear);
+    const endYearValue = endYear.getAttribute('value');
+    expect(endYearValue).toEqual('2022');
+  });
+
+  it('should save successfully when checking box and saving', async () => {
```

Comment:
I think either this test should sit inside the "_given time entries are to be persisted_" describe block or it could sit along with the above test in a new describe block around checkbox related stuff.
There are a ton of tests in this suite and would be good to keep them organised.

Reply:
Moved relevant tests into their own section